From ddf5ea4e8a95d40dc30b49cb6551e2a23f76eba4 Mon Sep 17 00:00:00 2001
From: Ioana Pandele <ioanap@google.com>
Date: Mon, 14 Oct 2024 13:27:10 +0000
Subject: [PATCH] [Passwords] Fix crash in Chrome sign in flow

Some enterprise flows crash when the user attempts to sign in to
Chrome, because there is no PasswordClientUIDelegate attached to the
WebContents. This broke in M129.

Bug: 372873259,372997985
Change-Id: I8054c742dd7fa786f4e5e3a02e7d13d9e75f780d
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5928687
Commit-Queue: Ioana Pandele <ioanap@chromium.org>
Reviewed-by: Maria Kazinova <kazinova@google.com>
Cr-Commit-Position: refs/heads/main@{#1368214}
---

--- a/chrome/browser/password_manager/chrome_password_manager_client.cc
+++ b/chrome/browser/password_manager/chrome_password_manager_client.cc
@@ -370,6 +370,11 @@
 #else
   PasswordsClientUIDelegate* manage_passwords_ui_controller =
       PasswordsClientUIDelegateFromWebContents(web_contents());
+  // TODO(crbug.com/372873259): This suddenly started being called/being null,
+  // in the Chrome sign-in flow in M129. Find out why.
+  if (!manage_passwords_ui_controller) {
+    return false;
+  }
   if (update_password) {
     manage_passwords_ui_controller->OnUpdatePasswordSubmitted(
         std::move(form_to_save));
